src/devices/Seesaw/SeesawBase.cs

```diff
@@ -0,0 +1,176 @@
+
+            if(ReadByte(Seesaw_Module.Status, Seesaw_Function.Status_Hw_Id) != Sessaw_Hardware_Id)
+            {
+                throw new NotSupportedException($"The hardware on I2C Bus {I2cDevice.ConnectionSettings.BusId}, Address 0x{I2cDevice.ConnectionSettings.DeviceAddress:X2} does not appear to be an Adafruit SeeSaw module");
+            }
+
+            _options = GetOptions();
+            Version = GetVersion();
+        }
+
+        /// <summary>
+        /// Get the firmware version of the Seesaw board. 
+        /// </summary>
+        /// <returns>Returns the seesaw version.</returns>
+        protected UInt32 GetVersion()
+        {
+            byte[] buffer = Read(Seesaw_Module.Status, Seesaw_Function.Status_Version, 4);
+
+            return (((UInt32)buffer[0] << 24) | ((UInt32)buffer[1] << 16) | ((UInt32)buffer[2] << 8) | (UInt32)buffer[3]);
+        }
+
+        /// <summary>
+        /// Gets the options/modules present on the Seesaw board
+        /// </summary>
+        /// <returns>Returns 32 bit integer where the bits represent the options/modules present on the Seesaw board.</returns>
+        protected UInt32 GetOptions()
+        {
+            byte[] buffer = Read(Seesaw_Module.Status, Seesaw_Function.Status_Options, 4);
+
+            return (((UInt32)buffer[0] << 24) | ((UInt32)buffer[1] << 16) | ((UInt32)buffer[2] << 8) | (UInt32)buffer[3]);
+        }
+
+        /// <summary>
+        /// Write a byte to the I2cDevice connected to the Seesaw board. 
+        /// </summary>
+        /// <param name="moduleAddress">An Seesaw_Module enum that represents the module that we are writing to.</param>
+        /// <param name="functionAddress">An Seesaw_Function enum that represents the Seesaw function to be called.</param>
+        /// <param name="value">The byte value ro be send as a parameter to the Seesaw device.</param>
+        protected void WriteByte(Seesaw_Module moduleAddress, Seesaw_Function functionAddress, byte value)
+        {
+            Write(moduleAddress, functionAddress, new byte[] { value });
+        }
+
+        /// <summary>
+        /// Write a series of bytes to the I2cDevice connected to the Seesaw board. 
+        /// </summary>
+        /// <param name="moduleAddress">An Seesaw_Module enum that represents the module that we are writing to.</param>
+        /// <param name="functionAddress">An Seesaw_Function enum that represents the Seesaw function to be called.</param>
+        /// <param name="data">The byte array containg data to be send as a parameter or parameters to the Seesaw device.</param>
+        protected void Write(Seesaw_Module moduleAddress, Seesaw_Function functionAddress, byte[] data)
+        {
+            const int StackThreshold = 512;
+
+            Span<byte> buffer = data.Length < StackThreshold ? stackalloc byte[data.Length + 2] : new byte[data.Length + 2];
+
+            for (int i = 0; i < data.Length; buffer[i + 2] = data[i], i++) ;
```

Comment:
put `buffer[i + 2] = data[i]` in the body of the `for` loop

Reply:
Done in the latest PR update by changes to just use Span<byte> as input.